src/mixin/entity-mixin-lit.js

```diff
@@ -85,11 +52,27 @@ export const EntityMixinLit = superclass => class extends superclass {
 		}
 	}
 
-	__onHrefChange(href, token) {
-		dispose(this._entity); // Make sure the entity is cleaned up before setting a new one.
+	_getEntity() {
+		dispose(this.entity);
 		if (typeof this._entityType === 'function') {
-			entityFactory(this._entityType, href, token, entity => {
-				this._entity = entity;
+			let pendingResolve;
+			var pendingPromise = new Promise(function(resolve) {
+				pendingResolve = resolve;
+			});
+
+			const pendingEvent = new CustomEvent('pending-state', {
```

Comment:
Do we want to align on a D2L specific event, `d2l-pending-state`?

Reply:
I was using  `pending-state` for now because that is what the `@brightspace-ui/core` `runAsync` directive is using and I thought it might be good to keep the PendingContainer usable with runAsync too.  But I'm ok to switch it to use `d2l-pending-state` if we prefer.

Reply:
Yeah, I don't feel strongly about it.  I don't think anything is using `pending-state` from `run-async` so we could update them both to be consistent.  The original `pending-state` event name was from Justin's PR which is unlikely to be merged.